noxfile.py

```diff
@@ -73,8 +75,10 @@ def run_coverage(session):
 def run_integration(session):
     """Run the integration tests."""
     set_environment_variables(PYBAMM_ENV, session=session)
+    # TODO: remove this when PyBaMM moves to pyproject.toml
+    session.install("--upgrade", "pip", "setuptools", "wheel", silent=False)
     session.install("-e", ".[all]", silent=False)
-    if sys.platform == "linux":
+    if sys.platform == "linux" and sys.version_info < (3, 12):
```

Comment:
We would update this too once done with #3550, right?

Reply:
Yes, will update this branch soon